src/tools/rpc.rs

```diff
@@ -0,0 +1,224 @@
+use std::{
+    fmt::{self, Display, Formatter},
+    time::Duration,
+};
+
+use data_encoding::{BASE32_NOPAD, BASE64};
+use fmt::Debug;
+use reqwest::{header, Client};
+use serde::{
+    de::{Error, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use tokio::time::{error::Elapsed, sleep};
+
+use crate::protocol::constants::USER_AGENT;
+
+/// Timeout time for RPC requests.
+const RPC_TIMEOUT: Duration = Duration::from_secs(10);
+
+/// Returns a block for a provided round.
+pub async fn wait_for_block(rpc_addr: &str, round: u64) -> Result<EncodedBlockCert, Elapsed> {
+    // Algod V1 documentation states that the round format is 'integer (int64)',
+    // but it's actually an int64 integer encoded in base36.
+    let round = radix_fmt::radix_36(round).to_string();
+
+    tokio::time::timeout(RPC_TIMEOUT, async move {
+        loop {
+            if let Ok(rsp) = get_block(rpc_addr, &round).await {
+                if rsp.error_for_status_ref().is_err() {
+                    tracing::trace!("invalid status for the response {:?}", rsp);
+                    continue;
+                }
+                tracing::info!("correct status for the response {:?}", rsp);
+
+                let block = rmp_serde::from_slice(&rsp.bytes().await.unwrap()).unwrap();
+                tracing::info!("block data {:?}", block);
+                return Ok(block);
+            }
+            sleep(Duration::from_secs(1)).await;
+        }
+    })
+    .await?
+}
+
+async fn get_block(
+    rpc_addr: &str,
+    round: &str,
+) -> anyhow::Result<reqwest::Response, reqwest::Error> {
+    Client::new()
```

Comment:
Add a comment about how this is exactly the same request our synth node receives from the node.

Reply:
do we even need to perform a handshake if we're making a call using a `reqwest` client which is not a part of the node?

Reply:
Nope, what I did here is imitated the behaviour of the real node

So my next test could be "ask for blocks without the handshake"

Reply:
hmm, if the true node behaves like this (i.e. sends HTTP requests on a different port than the handshake), we should probably bundle the synthetic node with the `reqwest::Client`

Reply:
> (i.e. sends HTTP requests on a different port than the handshake),

This is exactly what happens. If the node initiates the handshake, we end up upgrading the WebSocket connection but receive dozens of requests in `protocol/handshake.rs`, which are all simple GET block requests, but all of those get a parsing error since we only expect a handshake HTTP request there...

Reply:
> we should probably bundle the synthetic node with the reqwest::Client

Yeah, I was wondering if that would make sense.

Do we wanna implement something that will reply with block responses on the node's block requests?

Reply:
if the other node expects this and further behavior depends on it, we can do it in a follow-up, yes